### PR TITLE
add support for max, min, last, first operators in aggregation

### PR DESCRIPTION
**Pull request by drorasaf**
add support for max, min, last, first operators in aggregation

### CODE REVIEW

**Comment by srinivasreddy** [comment]
Could you pls resolve conflicts

**Comment by drorasaf** [comment]
I have to mention that test_aggregate10 from the recent commit, outputs:
TypeError: unorderable types: dict() < dict(),
Both on the origin and after the merge

**Comment by JulieRossi** [comment]
Hi,

This commit comes from a PR I issued. But I don't have that error when I run tests (test_aggregate10 passes).
Did you find out why this error happens ?

Let me know if I can help.

**Comment by drorasaf** [comment]
I get this error in python3. 
I do not get this error in python2.
This issue is related to differences between python3 and python2.
Python2 allows comparing of dictionaries but incorrectly

**Comment by JulieRossi** [comment]
Ok, indeed, thanks !
I figured out how to fix it.
I think I will wait for this PR to be merged and then submit a new one to fix this so it doesn't get confusing.

File: mongomock/collection.py [hunk 1]
```diff
@@ -1417,19 +1418,44 @@ def aggregate(self, pipeline, **kwargs):
                                             new_doc = False
                                             break
 
-                                    current_val = doc_dict.get(field, 0)
                                     from_field = key.replace('$', '')
                                     if func == "$sum":
+                                        current_val = doc_dict.get(field, 0)
                                         for doc in group_list:
```
Comment (srinivasreddy) [inline mongomock/collection.py]: `doc_dict[field] = doc_dict.get(field,0) + sum([doc.get(from_field,0) for doc in group_list])`

Comment (srinivasreddy) [inline mongomock/collection.py]: Could you please try for the all functions? does it work?

**Comment by srinivasreddy** [comment]
Thanks

**Pull request merged**